### PR TITLE
feat: migrate to gh-workflow-tailcall

### LINKED ISSUE #33: Migrate to `gh-workflow-tailcall`
More over here:   https://github.com/tailcallhq/gh-workflow/tree/main/crates/gh-workflow-tailcall

### PR DESCRIPTION
**Pull request by mehul-m-prajapati**
/claim #33

fixes #33

### CODE REVIEW

**Comment by tusharmath** [comment]
Please run cargo test and push the generated changes. Moving to draft to reduce noise and improve CI efficiency. Once you are ready just mark it as "ready to review". Feel free to give a shoutout on the #contributors channel on Discord if you want immediate attention.

**Comment by mehul-m-prajapati** [comment]
@tusharmath : Thanks for your review. I have added CI file which is generated by `cargo test`

File: tests/ci.rs [hunk 1]
```diff
@@ -1,69 +1,10 @@
...
-use toolchain::Toolchain;
+use gh_workflow_tailcall::*;
 
 #[test]
-fn ci() {
-    let flags = RustFlags::deny("warnings");
+fn generate_ci_workflow() {
+    let workflow = Workflow::default()
+        .auto_release(true)
+        .name("Build and Test".into());
```
Comment (tusharmath) [inline tests/ci.rs:7]:
```suggestion
```

`name` doesn't need to be set.

Comment (mehul-m-prajapati) [reply tests/ci.rs:7]: Okay , I will remove that and push it within some time

**Review by tusharmath — APPROVED** [review]

**Pull request merged**